Simplified getting started section

davecranwell opened the pull request:
These changes

* depend on the removal of the sass dependency
* assume the user is reasonably competent with django, therefore doesn't explain installing python etc.
* assumes prior merge of #1307 which adds a page in this section on the demo site

I expect to have to rebase once #1307 is merged.

kaedroho commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,42 @@
 Getting started
 ===============
 
+Wagtail is built on Django, so this document assumes you've already got the essentials installed. But if not, those essentials are:
```
Maybe refer to "Django" here as "the Django web framework" and make it a link

tomdyson commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,42 @@
+ * `Virtualenv <https://virtualenv.pypa.io/en/latest/installation.html>`_
+
+
+With the above installed and enabled, the quickest way to install wagtail is::
```
wagtail -> Wagtail

tomdyson commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,42 @@
+ * `Python <https://wiki.python.org/moin/BeginnersGuide/Download>`_
+ * `pip <https://pip.pypa.io/en/latest/installing.html>`_ (Note that pip is included by default with Python 2.7.9 and later and Python 3.4 and later)
+
+We'd also strongly recommend:
```
"We also recommend"

tomdyson commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,42 @@
+Your site is now accessible at ``http://localhost:8000``, with the admin backend available at ``http://localhost:8000/admin/``.
+
+
+There are a few optional packages which are not installed by default but are recommended to improve performance or add features to Wagtail:
```
Is it 'a few' or just two?

tomdyson commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,42 @@
+With the above installed and enabled, the quickest way to install wagtail is::
+
+    pip install wagtail
+
```
We probably need a callout box here about possible Pillow installation issues and instructions or links for OS X / Debian / Ubuntu / Windows.

tomdyson replied:
Perhaps just link to http://pillow.readthedocs.org/en/latest/installation.html#os-x-installation and http://pillow.readthedocs.org/en/latest/installation.html#windows-installation etc.

davecranwell replied:
What are those installations issues? I've tried following these instructions myself on osx (python 2.7.5) and not seen any errors yet.

kaedroho replied:
You won't get an error. Just a Pillow library that doesn't work

kaedroho commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,46 @@
+
+    Wagtail depends on the Pillow package for image manipulation. On some platforms Pillow requires additional steps to get images working in Wagtail. `More information here <http://pillow.readthedocs.org/en/latest/installation.html#os-x-installation>`_
+
+Once installed, Wagtail provides a command similar to Django's ``django-admin startproject``::
```
Maybe mention in this sentence somewhere that this will start a new project. Keep in mind many readers of this doc will not have used Django before.

kaedroho commented:
Could you rebase this. It appears that you've done lots of this cleanup in https://github.com/torchbox/wagtail/pull/1307 as well

davecranwell commented:
Done

kaedroho commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,46 @@
+ * `Python <https://wiki.python.org/moin/BeginnersGuide/Download>`_
+ * `pip <https://pip.pypa.io/en/latest/installing.html>`_ (Note that pip is included by default with Python 2.7.9 and later and Python 3.4 and later)
+
+We'd also recommend:
```
We recommend using ``virtualenv``, which provides isolated Python environments

davecranwell replied:
Shall we just remove the virtualenv refs? it seems like one of those things a seasoned django person would understand already anyway. The fact we don't mention it again, having mentioned it here, is a bit weird too.

kaedroho replied:
Shall we require that everyone knows Django/Python before they start learning Wagtail?

I would really like it if the getting started docs worked well for non Python/Django devs too (which I think would be most people).

davecranwell replied:
We're responsible for documenting this software not every prerequisite bit of knowledge, so yes we do require Django/Python understanding as we have to draw the line somewhere. However if you're aware of a suitable tutorial to learn Python from scratch which we can link at the start of this document, let me know, but we definitely shouldn't be documenting any of that ourselves.

zerolab replied:
FWIW, I think http://www.diveintopython.net/ is a good start.

kaedroho replied:
By the way, I'm not opposed to removing the references to virtualenv. But at the same time, I don't want us to assume that everyone who reads this doc is a Python/Django wizard already.

kaedroho commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,46 @@
+
+.. note::
+
+    Wagtail depends on the Pillow package for image manipulation. On some platforms Pillow requires additional steps to get images working in Wagtail. `More information here <http://pillow.readthedocs.org/en/latest/installation.html#os-x-installation>`_
```
Issues with Pillow must be resolved before running ``pip install wagtail`` or Pillow may need to be reinstalled.

kaedroho replied:
Could we just say "Install these two packages" instead of sending people to read a big long document?

But at least point to a more generic part of that doc (currently pointing to "OS X Installation" and not everyone uses OS X!)

davecranwell replied:
What are those two packages? Happy to add that if it's simpler. 

It's not really a link to OSX, but rather a link to the first in a list of platform-specific installation instructions. OSX happens to come first so is the best heading to attach to.

kaedroho replied:
Debian/Ubuntu: libjpeg-dev and libz-dev
Red Hat/Cent OS/Fedora: libjpeg-devel and zlib-devel
Arch: libjpeg-turbo, zlib
Windows users should be OK as Pillow ships a wheel file for them

davecranwell replied:
And for OSX?

kaedroho replied:
Haven't a clue (I don't use macs). I don't think they have those in the app store

davecranwell replied:
Well this is my point and is my reason for linking off-site. If we're not personally in possession of the knowledge but a third party who actively maintains that knowledge far better than us is, isn't it just simpler to link off?

kaedroho replied:
Yep, definitely simpler for us to do that. But put it before the ``pip install wagtail`` to stop wagtail installing a broken version of Pillow.

Also, Linux users there are being told to install Pillow as a system package which don't work with virtual environments.

kaedroho commented on docs/howto/performance.rst:
```diff
@@ -33,6 +33,19 @@ Search
 
 Wagtail has strong support for `Elasticsearch <http://www.elasticsearch.org/>`_ - both in the editor interface and for users of your site - but can fall back to a database search if Elasticsearch isn't present. Elasticsearch is faster and more powerful than the Django ORM for text search, so we recommend installing it or using a hosted service like `Searchly <http://www.searchly.com/>`_.
 
+Once you've installed the Elasticsearch binaries, install the python connector library with::
```
Once the Elasticsearch server is running. Install the ``elasticsearch`` python module with.

kaedroho commented on docs/howto/performance.rst:
```diff
@@ -33,6 +33,19 @@ Search
+Once you've installed the Elasticsearch binaries, install the python connector library with::
+
+    pip install elasticsearch
+    ./manage.py update_index 
```
This command can't be run before configuring the backend in your settings

kaedroho commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,46 @@
 
+Wagtail is built on the `Django web framework <https://www.djangoproject.com/>`_, so this document assumes you've already got the essentials installed. But if not, those essentials are:
+
+ * `Python <https://wiki.python.org/moin/BeginnersGuide/Download>`_
```
What about https://www.python.org/downloads/?

kaedroho commented on docs/getting_started/index.rst:
```diff
@@ -1,11 +1,46 @@
+
+With the above installed and enabled, the quickest way to install Wagtail is::
+
+    pip install wagtail
```
``sudo pip install wagtail`` for most unix users.

tomdyson replied:
depends if you're in a virtualenv or not, no?

kaedroho replied:
Yep, but are we recommending that Wagtail should be installed in a virtualenv or system wide?

We're pointing out that we need Virtualenv. No indication where or why though.

davecranwell replied:
Wouldn't a note "sudo may be required if you experience errors" be sufficient?

kaedroho replied:
sudo isn't the answer to all problems though. Why not explain that if you're using a unix box and want to install it system-wide, use sudo?

kaedroho commented on docs/howto/performance.rst:
```diff
@@ -33,6 +33,23 @@ Search
 
 Wagtail has strong support for `Elasticsearch <http://www.elasticsearch.org/>`_ - both in the editor interface and for users of your site - but can fall back to a database search if Elasticsearch isn't present. Elasticsearch is faster and more powerful than the Django ORM for text search, so we recommend installing it or using a hosted service like `Searchly <http://www.searchly.com/>`_.
 
+Once the Elasticsearch server is installed and running. Install the ``elasticsearch`` python module with::
```
python => Python

kaedroho commented:
:+1:

davecranwell commented:
Lets just merge now and tweak with tips etc later.

kaedroho commented:
Some warnings:
```
/home/karl/projects/wagtail/docs/getting_started/demo_site.rst:6: WARNING: unknown document: creating_your_project
/home/karl/projects/wagtail/docs/howto/settings.rst:47: WARNING: undefined label: wagtail_site_admin (if the link has no caption the label must precede a section header)
/home/karl/projects/wagtail/docs/reference/project_template.rst:51: WARNING: unknown document: ../getting_started/using_vagrant
```

SalahAdDin commented:
@zerolab offtopic, i didn't know something about docutils, thak you :D

davecranwell commented:
* getting_started/demo_site.rst:6  - fixed
 * howto/settings.rst:47 - this appears to be a result of your partial removal of the Sites docs https://github.com/torchbox/wagtail/commit/9139e6454f53dafe5e4757130cff36f9a12377e1.Fixed.
 * reference/project_template.rst:51 - Since vagrant is no longer ref'd in the getting started i've removed the line, however refs to Vagrant in the project template are still relevant as at time of merging this PR (today, hopefully) the project template still contains vagrant files.

kaedroho commented:
Merged in bd45a12efd2ffebd2583d739c7cd1f8cd18a9f8f

zerolab commented:
:shipit:
